.github/workflows/CI.yml

```
@@ -37,6 +37,6 @@ jobs:
       - uses: julia-actions/julia-buildpkg@v1
       - uses: julia-actions/julia-runtest@v1
       - uses: julia-actions/julia-processcoverage@v1
-      - uses: codecov/codecov-action@v3
+      - uses: codecov/codecov-action@v4
         with:
```

v4 does not support tokenless uploads (see e.g. https://github.com/codecov/codecov-action?tab=readme-ov-file#v4-release). Someone with admin permissions has to add the codecov token as a Github secret and then we can use
```suggestion
        with:
          token: ${{ secrets.CODECOV_TOKEN }} # required
```